sim/battle-stream.ts

```diff
@@ -225,11 +225,12 @@ export class BattlePlayer {
 	readonly log: string[];
 	readonly debug: boolean;
 
-	constructor(playerStream: Streams.ObjectReadWriteStream, debug: boolean = false) {
+	constructor(
+		playerStream: Streams.ObjectReadWriteStream, debug: boolean = false, errorHandler?: (e: any) => void) {
 		this.stream = playerStream;
 		this.log = [];
 		this.debug = debug;
-		this.listen();
+		this.listen().catch(errorHandler || ((e: any) => { throw e; }));
```

Comment:
I'm not sure what this catch actually adds here? It just turns an unhandled promise rejection into an unhandled error.

Reply:
Oh, error handler. Like I said on Discord, I think the correct call here is to make the stream itself forward the error.